tests/network/expose.go

```diff
@@ -225,8 +292,69 @@ var _ = SIGDescribe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:c
 				Expect(endpoint.Ports).To(ContainElement(k8sv1.EndpointPort{Name: "port-3", Port: 82, Protocol: "UDP"}))
 				Expect(endpoint.Ports).To(ContainElement(k8sv1.EndpointPort{Name: "port-4", Port: 1500, Protocol: "UDP"}))
 			},
-				table.Entry("[test_id:1533] over default IPv4 IP family", k8sv1.IPv4Protocol),
-				table.Entry("over IPv6 IP family", k8sv1.IPv6Protocol),
+				table.Entry("[test_id:1533] over default IPv4 IP family", ipv4),
+				table.Entry("over IPv6 IP family", ipv6),
+				table.Entry("over dual stack, primary ipv4", dualIPv4Primary),
+				table.Entry("over dual stack, primary ipv6", dualIPv6Primary),
+			)
+		})
+
+		Context("Expose ClusterIP service with IPFamilyPolicy", func() {
+			const serviceNamePrefix = "cluster-ip-with-ip-family-policy"
+
+			var serviceName string
+			var vmiExposeArgs []string
+
+			BeforeEach(func() {
+				serviceName = randomizeName(serviceNamePrefix)
+
+				vmiExposeArgs = []string{
+					expose.COMMAND_EXPOSE,
+					"virtualmachineinstance", "--namespace", tcpVM.GetNamespace(), tcpVM.GetName(),
+					"--name", serviceName,
+				}
+			})
+
+			table.DescribeTable("Should expose a ClusterIP service with the correct IPFamilyPolicy", func(ipFamiyPolicy k8sv1.IPFamilyPolicyType) {
+				tests.SkipIfVersionBelow("IPFamilyPolicy property on a service requires v1.20 and above", "1.20")
+
+				if ipFamiyPolicy == k8sv1.IPFamilyPolicyRequireDualStack {
+					libnet.SkipWhenNotDualStackCluster(virtClient)
+				}
+
+				calcNumOfClusterIPs := func() int {
+					switch ipFamiyPolicy {
+					case k8sv1.IPFamilyPolicySingleStack:
+						return 1
+					case k8sv1.IPFamilyPolicyPreferDualStack:
+						isClusterDualStack, err := libnet.IsClusterDualStack(virtClient)
+						ExpectWithOffset(1, err).NotTo(HaveOccurred(), "should have been able to infer if the cluster is dual stack")
+						if isClusterDualStack {
+							return 2
+						}
+						return 1
+					case k8sv1.IPFamilyPolicyRequireDualStack:
+						return 2
+					}
+					return 0
+				}
+
+				vmiExposeArgs = append(vmiExposeArgs, "--ip-family-policy", string(ipFamiyPolicy))
+
+				By("Exposing the service via virtctl command")
+				virtctl := tests.NewRepeatableVirtctlCommand(vmiExposeArgs...)
+				Expect(virtctl()).To(Succeed(), "should expose a service via `virtctl expose ...`")
+
+				By("Getting back the service")
+				svc, err := virtClient.CoreV1().Services(tcpVM.Namespace).Get(context.Background(), serviceName, k8smetav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Validating the num of cluster ips")
+				Expect(len(svc.Spec.ClusterIPs)).To(Equal(calcNumOfClusterIPs()))
```

Comment:
@AlonaKaplan Shouldn't you use `assert.XFail` here with the reference to the 1.20 issue?

Reply:
This part of the dual stack service is working. Dual stack service gets two cluster ips.